File: composeApp/src/androidMain/kotlin/co/jarias/flexapp/ui/screens/bingo/card_setup/BingoCardSetupScreenState.kt
package co.jarias.flexapp.ui.screens.bingo.card_setup

import co.jarias.flexapp.domain.BingoCell

data class BingoCardSetupScreenState(
    val gameId: Long = 0,
    val cardNumbers: List<List<String>> = List(5) { List(5) { "" } },
    val currentColumn: Int = 0,
    val isLoading: Boolean = false,
    val isSaving: Boolean = false,
    val errorMessage: String? = null,
    val cardSaved: Boolean = false
) {
    val columnLabels = listOf("B", "I", "N", "G", "O")
    val columnRanges = listOf(1..15, 16..30, 31..45, 46..60, 61..75)

    val currentColumnLabel: String get() = columnLabels.getOrElse(currentColumn) { "" }
    val currentColumnRange: IntRange get() = columnRanges.getOrElse(currentColumn) { 1..15 }
}

File: iosApp/iosApp/Screens/bingo/card_setup/BingoCardSetupScreenState.swift
import Foundation
import shared

struct BingoCardSetupScreenState {
    var gameId: Int64 = 0
    var cardNumbers: [[String]] = Array(repeating: Array(repeating: "", count: 5), count: 5)
    var currentColumn: Int = 0
    var isLoading: Bool = false
    var isSaving: Bool = false
    var errorMessage: String? = nil
    var cardSaved: Bool = false
    
    let columnLabels = ["B", "I", "N", "G", "O"]
    let columnRanges = [1...15, 16...30, 31...45, 46...60, 61...75]
    
    var currentColumnLabel: String {
        guard currentColumn >= 0 && currentColumn < columnLabels.count else { return "" }
        return columnLabels[currentColumn]
    }
    
    var currentColumnRange: ClosedRange<Int> {
        guard currentColumn >= 0 && currentColumn < columnRanges.count else { return 1...15 }
        return columnRanges[currentColumn]
    }
}
